.github/workflows/publish.yml

```diff
@@ -77,22 +75,54 @@ jobs:
         with:
           fetch-depth: 0
           submodules: recursive
-      - uses: astral-sh/setup-uv@v5
+      - name: Install uv and set the python version to ${{ env.python-version }}
+        uses: astral-sh/setup-uv@v5
         with:
           python-version: ${{ env.python-version }}
-      - run: uv pip install build
-      - run: |
+      - name: Build package
+        run: |
           git reset --hard HEAD
-          python -m build
+          rm -f src/pynxtools/nomad/nxs_metainfo_package_*.json
+          uv run --extra nomad scripts/generate_package.py
+          uv build
       - name: Upload dist artifacts
         uses: actions/upload-artifact@v4
         with:
           name: dist-artifacts
           path: dist/
 
+  test-install:
+    name: Install built package
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install uv and set the python version to ${{ env.python-version }}
+        uses: astral-sh/setup-uv@v5
+        with:
+          python-version: ${{ env.python-version }}
+      - name: Download dist artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist-artifacts
+          path: dist/
+      - name: Install sdist
+        run: |
+          uv pip install dist/*.tar.gz
+      - name: Check that package JSON file exists
+        run: |
+          SITE_PACKAGES=$(python -c "import site; print(site.getsitepackages()[0])")
+          FILE_COUNT=$(find "$SITE_PACKAGES/pynxtools/nomad" -name 'nxs_metainfo_package_*.json' | wc -l)
+          if [ "$FILE_COUNT" -eq 0 ]; then
+            echo "ERROR: nxs_metainfo_package_*.json not found in installed package."
+            exit 1
+          else
+            echo "Found $FILE_COUNT JSON file(s) in package:"
```

Comment:
shouldn't we test if the found package(s) correspond(s) to the actual definitions submodule?